vpr/src/base/partition.h

```diff
@@ -0,0 +1,52 @@
+#ifndef PARTITION_H
+#define PARTITION_H
+
+#include "vtr_strong_id.h"
+#include "region.h"
+#include "partition_regions.h"
+
+/**
+ * @file
+ * @brief This file defines the data for a partition: a grouping of atoms that are constrained to a portion of an FPGA. A partition defines
+ * the atoms that are assigned to it, and the locations in which they can be placed. The locations are a union of rectangles in x, y,
+ * sub_tile space. Common cases include a single rectangle or a single x, y, sub_tile (specific location). More complex partitions
+ * with L, T or other shapes can be created with a union of multiple rectangles.
+ */
+
+//Type tag for PartitionId
+struct partition_id_tag;
+
+//A unique identifier for a partition
+typedef vtr::StrongId<partition_id_tag> PartitionId;
+
+class Partition {
+  public:
+    PartitionId get_partition_id();
+    void set_partition_id(PartitionId _part_id);
+
+    std::string get_name();
+    void set_name(std::string _part_name);
+
+    //add an id to the atom_blocks vector
+    void add_to_atoms(AtomBlockId atom_id);
+
+    //check if a given atom is in the partition
+    bool contains_atom(AtomBlockId atom_id);
+
+    //get the atom_blocks vector
+    std::vector<AtomBlockId> get_atoms();
```

Comment:
This returns a new copy of the `std::vector`?  I assume this not the intention?  Better to have this return either `const std::vector<AtomBlockId> &` or `vtr:array_view<const AtomBlockId>`?

Method should be const?

Reply:
Good point. Yes, should be const, and should just return a reference.  Haven't used array_view; may be more appropriate still.